covdefaults.py

```diff
@@ -58,22 +59,35 @@ def _plat_impl_pragmas():  # type: () -> List[str]
 
 
 class CovDefaults(CoveragePlugin):
-    def __init__(self, subtract_omit: str = '') -> None:
+    def __init__(
+        self,
+        subtract_omit: str = '',
+        installed_package: str = '',
+    ) -> None:
         self._subtract_omit = subtract_omit.split()
+        self._installed_package = self._parse_inst_pkg(installed_package)
+
+    @staticmethod
+    def _parse_inst_pkg(installed_package: str) -> Dict[str, Optional[str]]:
+        result: Dict[str, Optional[str]] = {}
+        for entry in installed_package.split(' '):
+            if entry:
+                key, value = entry, None
+                if entry.count(':') == 1:
+                    key, value = entry.split(':')
+                result[key] = value
+        return result
 
     def configure(self, config: CoverageConfig) -> None:
-        for k, v in OPTIONS:
-            config.set_option(k, v)
+        self._set_source(config)
+        for opt_k, opt_v in OPTIONS:
+            config.set_option(opt_k, opt_v)
         for k, v in EXTEND:
             before = set(config.get_option(k) or ())
             before.update(v)
             config.set_option(k, sorted(before))
 
-        # subtract omit settings if requested
-        if self._subtract_omit:
-            omit = set(config.get_option('run:omit'))
-            omit.difference_update(self._subtract_omit)
-            config.set_option('run:omit', sorted(omit))
+        self.fix_omit(config)
```

Comment:
I prefer bottom-up functions instead of top-down functions (so move the implementations of these *above* the main `configure` function)

this should also be `_fix_omit`